core/templates/components/summary-tile/lesson-card.component.ts

```diff
@@ -0,0 +1,163 @@
+  @Input() topic!: string;
+
+  desc!: string;
+  imgColor!: string;
+  imgUrl!: string;
+  lessonUrl!: string;
+  progress!: number;
+  title!: string;
+  lessonTopic!: string;
+
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private assetsBackendApiService: AssetsBackendApiService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.story instanceof StorySummary) {
+      this.setStorySummary(this.story);
+    } else if (this.story instanceof CollectionSummary) {
+      this.setCollectionSummary(this.story);
+    } else {
+      this.setExplorationSummary(this.story);
+    }
+  }
+
+  setStorySummary(storyModel: StorySummary): void {
+    const completedStories = storyModel.getCompletedNodeTitles().length;
+    this.desc = storyModel.getTitle();
+    this.imgColor = storyModel.getThumbnailBgColor();
+    this.imgUrl = this.getStorySummaryThumbnailUrl({
+      filename: storyModel.getThumbnailFilename(),
+      id: storyModel.getId(),
+    });
+
+    const nextStory =
+      completedStories === storyModel.getAllNodes().length
+        ? completedStories - 1
+        : completedStories;
+
+    const lessonArgs = {
+      classroom: storyModel.getClassroomUrlFragment(),
+      topicFragment: storyModel.getTopicUrlFragment(),
+      currentStory: storyModel.getAllNodes()[nextStory],
+      story: storyModel.getUrlFragment(),
+    };
+    this.lessonUrl = this.getStorySummaryLessonUrl(lessonArgs);
+
+    this.title = `Chapter ${nextStory + 1}: ${storyModel.getNodeTitles()[nextStory]}`;
+    this.progress = Math.floor(
+      (completedStories / storyModel.getNodeTitles().length) * 100
+    );
+    this.lessonTopic = this.topic;
+  }
+
+  setCollectionSummary(collectionModel: CollectionSummary): void {
+    this.desc = collectionModel.objective;
+    this.imgColor = collectionModel.thumbnailBgColor;
+    this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+      collectionModel.thumbnailIconUrl
+    );
+    this.progress = 0;
+    this.title = collectionModel.title;
+    this.lessonUrl = `/collection/${collectionModel.id}`;
+    this.lessonTopic = 'Collections';
+  }
+
+  setExplorationSummary(explorationModel: LearnerExplorationSummary): void {
+    this.desc = explorationModel.objective;
+    this.imgColor = explorationModel.thumbnailBgColor;
+    this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+      explorationModel.thumbnailIconUrl
+    );
+    this.progress = 0;
+    this.title = explorationModel.title;
+    this.lessonUrl = `/explore/${explorationModel.id}`;
+    this.lessonTopic = 'Community Lessons';
+  }
+
+  getStorySummaryThumbnailUrl(storyData: {
+    filename: string;
+    id: string;
+  }): string {
+    return this.assetsBackendApiService.getThumbnailUrlForPreview(
+      AppConstants.ENTITY_TYPE.STORY,
+      storyData.id,
+      storyData.filename
+    );
+  }
+
+  getStorySummaryLessonUrl(storyData: {
```

Comment:
You don't need a single object here, let this function accept multiple arguments, but not a single object with multiple fields

Reply:
Done.